### PR TITLE
Added message receiver

### PR DESCRIPTION
**Pull request by liamwilson-kainos**
### Summary

Eligibility service message receiver

https://dev.azure.com/defragovuk/DEFRA-FFC/_workitems/edit/135103

### CODE REVIEW

**Review by marcinmo-kainos — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -12,6 +12,14 @@ Optional:
...
 - Helm
 
+## Folder Structure
+
+The folder structure follows an approach of seperating functionality into use cases and keeping this 
+seperate from generic boilerplate code.
+
+`app/register-your-interest` contains all the functionality for the register your interest journey.
+`app/auto-elgibility` contains all the business logic for checking a farmers elgibility to gain access to Vet Vists.
+
```
Comment (AbidemiAdio) [inline README.md:22]: Message sample/structure or reference to message sample/structure should be in readme

Comment (liamwilson-kainos) [reply README.md:22]: Done! :)

File: app/config/index.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+const registerYourInterestConfig = require('../register-your-interest/config')
```
Comment (AbidemiAdio) [inline app/config/index.js:1]: What do you feel about having all app related config-files within one config-folder?

File: app/index.js [hunk 3]
```diff
@@ -1,13 +1,11 @@
...
-  console.log('Server running on %s', server.info.uri)
-}
+startServer()
+startMessageService()
 
-process.on('unhandledRejection', (err) => {
-  console.error(err)
-  process.exit(1)
+process.on('unhandledRejection', async (err) => {
+  await stopMessageService()
```
Comment (AbidemiAdio) [inline app/index.js:9]: We need to check and confirm that any unhandled exception should stop messaging service.

Comment (liamwilson-kainos) [reply app/index.js:6]: Done - this was a tricky test 😆

File: app/config/index.js (hunk 2 above)
Comment (liamwilson-kainos) [reply app/config/index.js:1]: Okay if we keep this the same so we can easily delete the register-your-interest folder once we are done with it?

**Review by AbidemiAdio — APPROVED** [review]
QED

**Pull request merged**